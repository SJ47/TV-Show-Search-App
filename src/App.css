*,
*::before,
*::after {
    box-sizing: border-box;
}

body {
    margin: 0;
    padding: 0;
    font-family: "Rubik", sans-serif;
}

:root {
    --primary-color: blueviolet;
    --secondary-color: orange;

    --box-shadow-heavy: 0 4px 8px 0 rgba(0, 0, 0, 0.9),
        0 6px 20px 0 rgba(0, 0, 0, 0.9);
    --box-shadow-light: 0 4px 8px 0 rgba(0, 0, 0, 0.2),
        0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.container {
    display: flex;
    position: sticky;
    top: 0;
}

.header {
    padding: 0.3em;
    text-align: center;
    width: 100%;
    background-image: url(../src/images/tv-shows-logo.jpg);
    background-size: contain;
    margin-top: 0;
    padding-top: 0.5em;
}

.header-text {
    font-size: 1.5em;
    padding: 0 1em;
    color: white;
    background: var(--primary-color);
    border-radius: 25px;
    margin-top: 0.3em;
    box-shadow: var(--box-shadow-heavy);
}

form {
    padding-bottom: 1em;
}

.submit-btn {
    background: var(--primary-color);
    color: white;
    padding: 0.5em 1.5em;
    margin-left: 1em;
    font-size: 0.55em;
    border: none;
    outline: none;
    border-radius: 25px;
    cursor: pointer;
    width: 150px;
    box-shadow: var(--box-shadow-light);
}

.submit-btn:hover {
    font-weight: bolder;
    box-shadow: var(--box-shadow-heavy);
}

.input-search {
    padding: 0.5em 1.5em 0.5em 1em;
    font-size: 0.5em;
    border: 3px dashed grey;
    border-radius: 25px;
    outline: none;
    box-shadow: var(--box-shadow-light);
}

.input-search:focus {
    border: 3px dashed var(--primary-color);
    box-shadow: var(--box-shadow-heavy);
}

.tv-container {
    display: grid;
    grid-template-columns: 3fr 400px;
    grid-template-rows: 250px auto;
    gap: 1em;
    margin: 1em;
}

.favouritesContainer {
    grid-column-start: 1;
    grid-column-end: 2;
    text-align: center;
    box-shadow: var(--box-shadow-light);
}

.favouritesContainer::-webkit-scrollbar {
    display: none;
}

.favContainer {
    display: flex;
    overflow-x: auto;
    width: 80%;
    margin: auto;
    justify-items: center;
}

.favouriteShowsTitle {
    padding-bottom: 0;
}

.fav-image {
    margin: 1em;
    cursor: pointer;
}

.fav-image:hover {
    box-shadow: var(--box-shadow-light);
}

.left-column {
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 3;
    box-shadow: var(--box-shadow-light);
    height: fit-content;
}

.search-results-header {
    margin-left: 1.5em;
    padding-bottom: 0;
    text-align: center;
}

.right-column {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 3;
    height: fit-content;
    text-align: center;
    width: 100%;
    box-shadow: var(--box-shadow-light);
}

ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: left;
}

li {
    list-style: none;
    cursor: pointer;
    padding: 0.5em;
}

.card-img {
    opacity: 0.9;
    box-shadow: var(--box-shadow-light);
}

.card-img:hover {
    opacity: 1;
    box-shadow: var(--box-shadow-heavy);
}

h2 {
    margin: 0em;
    padding: 0.9em;
}

hr {
    border: 3px solid var(--secondary-color);
    border-radius: 50%;
    margin-bottom: 1em;
    width: 75%;
}

.selected-card-title {
    background: var(--primary-color);
    color: white;
}

.rating-fav-text-container {
    display: flex;
    justify-content: space-between;
}

.rating-text {
    padding-left: 1em;
    margin-bottom: 0;
    padding-bottom: 0;
    text-align: left;
    font-size: 1.5em;
}
.favourite-text {
    padding-right: 0.8em;
    margin-bottom: 0;
    padding-bottom: 0;
    font-size: 1.5em;
}

.rating-fav-circle-container {
    display: flex;
    justify-content: space-between;
}

.rating {
    background: var(--primary-color);
    color: white;
    padding: 1em;
    font-weight: bold;
    border-radius: 50%;
    font-size: 1.2em;
    margin: 1em;
    margin-left: 1.5em;
    width: auto;
    height: 100%;
}

.fa-plus-circle,
.fa-minus-circle {
    color: var(--primary-color);
    font-size: 4em;
    padding-right: 0.6em;
    margin-top: 0.25em;
    cursor: pointer;
}

.summary-text {
    text-align: left;
    margin: 0em 1em;
}

.summary-text-title {
    padding: 0 0 0.3em 0;
}
