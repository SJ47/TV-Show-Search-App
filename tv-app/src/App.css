*,
*::before,
*::after {
    box-sizing: border-box;
}

body {
    margin: 0;
    padding: 0;
    /* font-family: "Lato", sans-serif; */
    font-family: "Rubik", sans-serif;
}

:root {
    /* --primary-color: #ffd230;
    --secondary-color: #000; */
    --primary-color: blueviolet;
    --secondary-color: orange;
}

#root {
    /* width: 90%; */
    /* margin: 1em auto 1em auto; */
    /* border: 5px dashed grey; */
    /* border-radius: 25px; */
}
.container {
    display: flex;
    /* width: 80%; */
}

.header {
    padding: 0.3em;
    text-align: center;
    width: 100%;
    /* font-size: 3.5em; */
    /* background: transparent; */
    /* background-image: url(../src/images/brush-stroke-banner-6.svg); */
    background-image: url(../src/images/tv-shows-logo.jpg);
    /* background-size: 100% 95%; */
    background-size: contain;
    /* background-repeat: no-repeat; */
    /* background-position: center; */
    /* opacity: 0.5; */
    margin-top: 0;
    padding-top: 0.5em;
    /* border-radius: 25px; */
    /* border-top-left-radius: 25px;
    border-top-right-radius: 25px; */
}

.header-text {
    font-size: 1.5em;
    padding: 0 1em;
    color: white;
    background: blueviolet;
    border-radius: 25px;
    margin-top: 0.3em;
}

form {
    /* background: orange; */
    padding-bottom: 1em;
}

.submit-btn {
    background: var(--primary-color);
    color: white;
    padding: 0.5em 1.5em;
    margin-left: 1em;
    /* background: orange; */
    font-size: 0.55em;
    border: none;
    outline: none;
    border-radius: 25px;
    cursor: pointer;
}

.input-search {
    padding: 0.5em 1.5em 0.5em 1em;
    font-size: 0.5em;
    border: 3px dashed grey;
    border-radius: 25px;
    outline: none;
}

.tv-container {
    display: grid;
    grid-template-columns: 2fr 1fr;
    grid-template-rows: 1fr auto;
    margin: 1em;
}

.favouritesContainer {
    grid-column-start: 1;
    grid-column-end: 3;
    text-align: center;
    border: 3px solid grey;
}

.favContainer {
    /* background-color: orange; */
    display: flex;
    flex-wrap: wrap;
}

.fav-image {
    margin: 1em;
    cursor: pointer;
}

.fav-image:hover {
    border: 3px solid blueviolet;
}

.left-column {
    margin-right: 1em;
    margin-top: 1em;
    border: 3px solid grey;
}

.search-results-header {
    margin-left: 1.5em;
    padding-bottom: 0;
    text-align: center;
}

.right-column {
    height: auto;
    margin-top: 1em;
    border: 3px solid grey;
}

ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: left;
}

li {
    list-style: none;
    cursor: pointer;
    padding: 0.5em;
}

img {
    /* border: 3px solid grey; */
    /* border: 3px solid blueviolet; */
    /* border-radius: 25px; */
}

.card-img {
    opacity: 0.5;
}

.card-img:hover {
    opacity: 1;
}

.selected-card {
    text-align: center;
    border: 1px solid lightgray;
    border-radius: 25px;
    padding-bottom: 1em;
    margin-right: 2em;
    height: fit-content;
}

h2 {
    /* font-size: 2em; */
    margin: 0em;
    padding: 0.9em;
    /* background-color: rgba(18, 118, 240, 0.5); */
    border-top-left-radius: 25px;
    border-top-right-radius: 25px;
}

hr {
    /* font-size: 3em; */
    border: 3px solid blueviolet;
    border: 3px solid orange;
    border-radius: 50%;
    margin-bottom: 1em;
    width: 75%;
}

.selected-card-title {
    background: orange;
    background: blueviolet;
    color: white;
    /* opacity: 1; */
}

.rating-fav-text-container {
    display: flex;
    justify-content: space-between;
}

.rating-text {
    padding-left: 1em;
    margin-bottom: 0;
    padding-bottom: 0;
    text-align: left;
    font-size: 1.5em;
}
.favourite-text {
    padding-right: 0.8em;
    margin-bottom: 0;
    padding-bottom: 0;
    text-align: left;
    font-size: 1.5em;
}

.rating-fav-circle-container {
    display: flex;
    justify-content: space-between;
}

.rating {
    background: blueviolet;
    color: white;
    padding: 1em;

    border-radius: 50%;
    font-size: 1.2em;
    margin: 1em;
    margin-left: 1.5em;
    width: 64px;
}

.favourite {
    background: blueviolet;
    color: white;
    padding: 1em;
    border-radius: 50%;
    font-size: 1.2em;
    margin: 1em;
    margin-right: 1.5em;
    width: 64px;
    cursor: pointer;
}

.search-results-header {
}
