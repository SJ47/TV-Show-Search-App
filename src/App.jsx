import React from 'react'

import './App.css';
import TvContainer from './containers/TvContainer';

function App() {
  return (
    <>
      <TvContainer />
    </>
  );
}

export default App;
