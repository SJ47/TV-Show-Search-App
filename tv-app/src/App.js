import './App.css';
import TvContainer from './containers/TvContainer';
import GameContainer from "./containers/TvContainer"

function App() {
  return (
    <>
      <TvContainer />
    </>
  );
}

export default App;
