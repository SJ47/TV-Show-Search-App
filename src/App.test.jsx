import { render } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('should render styles correctly', () => {
    const { container } = render(
        <App />
    );
    expect(container).toMatchSnapshot();
  });
});
