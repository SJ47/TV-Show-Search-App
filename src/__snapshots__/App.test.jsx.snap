// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render styles correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="header"
    >
      <span
        class="header-text"
      >
        TV Show Search
      </span>
      <hr />
      <form>
        <input
          class="input-search"
          placeholder="Search for TV show"
          type="text"
        />
        <input
          class="submit-btn"
          type="submit"
          value="Search"
        />
      </form>
    </h1>
  </div>
  <div
    class="tv-container "
  >
    <div
      class="favouritesContainer"
    >
      <h2
        class="favouriteShowsTitle"
      >
        Favourite Shows
      </h2>
      <hr />
      <div
        class="favContainer"
      />
    </div>
    <div
      class="left-column"
    >
      <h2
        class="search-results-header"
      >
        Search Results
      </h2>
      <hr />
      <ul>
        No results found
      </ul>
    </div>
  </div>
</div>
`;
